File: user-service/src/main/java/rs/edu/raf/dto/ClientRegisterDTO.java
package rs.edu.raf.dto;

import jakarta.validation.constraints.Pattern;

public record ClientRegisterDTO(
    @Pattern(regexp = "^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}$", message = "Email address must be valid!")
    String email,
    @Pattern(regexp = "^(\\+381|0)6\\d{7,8}$", message = "The phone number must start with +381 or 0, followed by 6 and then 7 or 8 digits!")
    String phoneNumber,
    String bankAccountNumber,

    String password,

    String code

) { }

File: user-service-kotlin/src/main/kotlin/org/example/userservicekotlin/dto/ClientRegisterDTO.kt
package org.example.userservicekotlin.dto

import jakarta.validation.constraints.Pattern

data class ClientRegisterDTO(
    @Pattern(regexp = "^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}$", message = "Email address must be valid!")
    val email: String,
    @Pattern(
        regexp = "^(\\+381|0)6\\d{7,8}$",
        message = "The phone number must start with +381 or 0, followed by 6 and then 7 or 8 digits!"
    )
    val phoneNumber: String,
    val bankAccountNumber: String,
    val password: String,
    val code: String
)
